pydantic/_internal/_generate_schema.py

```diff
@@ -538,7 +546,10 @@ def _typed_dict_schema(
             recursive_schema = core_schema.definition_reference_schema(obj_ref)
             self._recursion_cache[obj_ref] = recursive_schema
 
-        for field_name, annotation in _typing_extra.get_type_hints(typed_dict_cls, include_extras=True).items():
+        for field_name, annotation in get_type_hints_infer_globalns(
+            typed_dict_cls, localns=self.types_namespace, include_extras=True
```

Comment:
I confirmed that when you define a custom typeddict class, the module is the module in which it was defined, not the module of TypedDict, so this use of `get_type_hints_infer_globalns` seems sound to me.

Reply:
Let's hope they don't decided to erase that in the future like they do with the bases...